src/murmur/Messages.cpp

```diff
@@ -154,6 +155,73 @@ bool isChannelEnterRestricted(Channel *c) {
 	return false;
 }
 
+/// Process the Capabilities message, used to negotiate voice protocol.
+/// See the doc in Mumble.proto for details about this message.
+///
+/// @param uSource ServerUser that receives this message.
+/// @param msg The message itself.
+void Server::msgCapabilities(ServerUser *uSource, MumbleProto::Capabilities &msg) {
+	MSG_SETUP(ServerUser::Connected);
+	std::string protocol_list_str;
+	for (int i = 0; i < msg.supported_protocols_size(); ++i){
+		const std::string& protocol_type_string = msg.supported_protocols(i);
+
+		if (protocol_type_string.empty()) continue;
+		protocol_list_str.append(protocol_type_string);
+		protocol_list_str.append(", ");
+
+		VoiceProtocol vp(protocol_type_string);
+		uSource->supportedVoiceProtocols.append(vp);
+
+	}
+
+	log(uSource, QString("Client supports voice protocols: %1").arg(
+		QString::fromStdString(protocol_list_str.substr(0, protocol_list_str.length() - 2))));
```

Comment:
```suggestion
		QString::fromStdString(protocol_list_str.substr(0, protocol_list_str.length() - std::string(", ").size()))));
```
or even better: Define the concatenation sequence above as a constant and use that for the actual concatenation and for getting the size here.